[None][ci] move unittests to sub-directories

Funatiq opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Optional FlashInfer integration added with a central availability flag, PDL toggle via env var, and clearer install/error messaging; optimized ops exposed when available.

* **Tests**
  * PyTorch test suites reorganized from broad filters to more granular, explicit listings; several test paths corrected and per-suite durations updated.
  * A sampler test simplified to use a fixed configuration instead of external JSON.

* **Chores**
  * Updated test ownership mappings to adjust PR review routing.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

<!--
Please write the PR title by following this template:

**[JIRA ticket/NVBugs ID/GitHub issue/None][type] Summary**

Valid ticket formats:
  - JIRA ticket: [TRTLLM-1234] or [FOOBAR-123] for other FOOBAR project
  - NVBugs ID: [https://nvbugs/1234567]
  - GitHub issue: [#1234]
  - No ticket: [None]

Valid types (lowercase): [fix], [feat], [doc], [infra], [chore], etc.

Examples:
  - [TRTLLM-1234][feat] Add new feature
  - [https://nvbugs/1234567][fix] Fix some bugs
  - [#1234][doc] Update documentation
  - [None][chore] Minor clean-up
-->

## Description

- Move all unit test files to sub-directories.
- Update YAML files to run separate CI job for each unit test directory.
- Update test durations to ensure more even test distribution.

## Test Coverage

<!--
Please list clearly what are the relevant test(s) that can safeguard the changes in the PR. This helps us to ensure we have sufficient test coverage for the PR.
-->

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--reuse-test (optional)pipeline-id --disable-fail-fast --skip-test --stage-list "A10-PyTorch-1, xxx" --gpu-type "A30, H100_PCIe" --test-backend "pytorch, cpp" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx" --detailed-log --debug(experimental)]`

Launch build/test pipelines. All previously running jobs will be killed.

`--reuse-test (optional)pipeline-id ` *(OPTIONAL)* : Allow the new pipeline to reuse build artifacts and skip successful test stages from a specified pipeline or the last pipeline if no pipeline-id is indicated. If the Git commit ID has changed, this option will be always ignored. The DEFAULT behavior of the bot is to reuse build artifacts and successful test results from the last pipeline.

`--disable-reuse-test ` *(OPTIONAL)* : Explicitly prevent the pipeline from reusing build artifacts and skipping successful test stages from a previous pipeline. Ensure that all builds and tests are run regardless of previous successes.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-PyTorch-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-PyTorch-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--test-backend "pytorch, cpp"` *(OPTIONAL)* : Skip test stages which don't match the specified backends. Only support [pytorch, cpp, tensorrt, triton]. Examples: "pytorch, cpp" (does not run test stages with tensorrt or triton backend). Note: Does **NOT** update GitHub pipeline status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests in addition to running L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx".

`--detailed-log ` *(OPTIONAL)* : Enable flushing out all logs to the Jenkins console. This will significantly increase the log volume and may slow down the job.

`--debug ` *(OPTIONAL)* : **Experimental feature**. Enable access to the CI container for debugging purpose. Note: Specify exactly one stage in the `stage-list` parameter to access the appropriate container environment. Note: Does **NOT** update GitHub check status.

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`
and the `scripts/test_to_stage_mapping.py` helper.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

Funatiq committed:
[refactor] move unittests to sub-directories

- Moved all unittest files to sub-directories.
- Updated test lists in YAML files to include new test paths.

This commit improves the testing framework for unittest files, ensuring comprehensive validation of the new implementations.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq commented:
/bot run

Funatiq committed:
ci: remove duplicated tests

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #14161](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14161/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #10687](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/10687/) completed with status: 'FAILURE'

tensorrt-cicd commented:
[PR_Github #14279](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14279/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #10784](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/10784/) completed with status: 'FAILURE'

Funatiq committed:
feat: add custom attention operations to module exports

- Included `attn_custom_op_inplace` and `mla_custom_op_inplace` in the `__all__` list of the custom ops module.
- This change fixes unittest/_torch/compilation/test_add_norm.py::test_add_norm_fusion.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #14463](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14463/) [ run ] triggered by Bot

Funatiq committed:
feat: introduce flashinfer utility functions and refactor imports

- Added `flashinfer_utils.py` to manage flashinfer availability and PDL environment variable.
- Updated imports across multiple modules to source `IS_FLASHINFER_AVAILABLE` and `ENABLE_PDL` from the new utility file.
- This refactor improves code organization and maintains clarity regarding flashinfer's integration.
- This change fixes circular import issues in tensorrt_llm/_torch/attention_backend/__init__.py.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq committed:
fixup! [refactor] move unittests to sub-directories

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #14489](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14489/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #14474](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14474/) [ run ] completed with state `ABORTED`

tensorrt-cicd commented:
[PR_Github #14489](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14489/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #10944](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/10944/) completed with status: 'SUCCESS'

tensorrt-cicd commented:
[PR_Github #14593](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14593/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #14593](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14593/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #11024](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/11024/) completed with status: 'SUCCESS'

tensorrt-cicd commented:
[PR_Github #14780](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14780/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #14780](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14780/) [ run ] completed with state `FAILURE`
[/LLM/main/L0_MergeRequest_PR pipeline #11159](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/11159/) completed with status: 'FAILURE'

omera-nv reviewed (approved):
LGTM

tensorrt-cicd commented:
[PR_Github #14809](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14809/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #11181](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/11181/) completed with status: 'FAILURE'

tensorrt-cicd commented:
[PR_Github #14820](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14820/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #11192](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/11192/) completed with status: 'FAILURE'

Funatiq commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #14829](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14829/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #14829](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14829/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #11200](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/11200/) completed with status: 'FAILURE'

liji-nv reviewed (changes requested):
Please update CODEOWNERS file accordingly.

```
/tests/unittest/_torch/test_custom_ops.py @NVIDIA/trt-llm-torch-graph-compiler
/tests/unittest/_torch/test_autotuner.py @NVIDIA/trt-llm-torch-graph-compiler
```

Funatiq committed:
refactor: simplify test_trtllm_sampler by removing unused test_case fixture

- Removed the test_case fixture and its associated JSON loading, simplifying the test setup.
- Hardcoded sampling parameters for clarity and consistency in the test execution.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq committed:
chore: update CODEOWNERS to reflect new test file locations

- Changed paths for `test_custom_ops.py` and `test_autotuner.py` to their new sub-directory locations.
- This update ensures proper ownership and visibility for the relocated test files.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #14962](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14962/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #14962](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14962/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #11296](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/11296/) completed with status: 'FAILURE'

Funatiq commented:
/bot run --disable-fail-fast

Funatiq commented:
/bot run

Funatiq committed:
chore: update test durations for unittests

- Added new test duration entries for various unittest cases in the `_torch` module.
- Adjusted existing entries for clarity and consistency in naming conventions.
- This update enhances the tracking of test performance metrics.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #15096](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15096/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #11398](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/11398/) completed with status: 'SUCCESS'

Funatiq commented:
/bot run

tburt-nv reviewed (approved):
Only reviewing CODEOWNERs here, looks good to me.

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #15180](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15180/) [ run ] triggered by Bot

Funatiq commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #15817](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15817/) [ run ] triggered by Bot
